src.ts/dynamics/rigid_body_set.ts

```diff
@@ -102,14 +117,21 @@ export class RigidBodySet {
      * @param multibodyJoints - The set of multibody joints that may contain joints attached to the removed rigid-body.
      */
     public remove(handle: RigidBodyHandle, islands: IslandManager, colliders: ColliderSet, impulseJoints: ImpulseJointSet, multibodyJoints: MultibodyJointSet) {
-        this.raw.remove(handle, islands.raw, colliders.raw, impulseJoints.raw, multibodyJoints.raw)
+        // remove owned colliders
+        while (this.raw.rbNumColliders(handle) > 0) {
+            const colliderHandle = this.raw.rbCollider(handle, 0);  // do not use forEach to remove because the length of the list in the Rust will be changed.
```

Comment:
Are we safe from infinite loop here? If Rust decreases the length, then you can use a normal `for (i = max, i--)` loop.

Reply:
> Are we safe from infinite loop here? If Rust decreases the length, then you can use a normal `for (i = max, i--)` loop.

1, i think it's safe unless Rust will return `Number.POSITIVE_INFINITY`, others all will fail no matter it will be `undefined / null / NaN / string`.
2, in case if the list will be increased somehow somewhere while looping (i think no chance), this loop won't have bug than the fixed loop, it will finally remove all the colliders even dynamiclly added colliders somehow.

i first tired with the `--` version and changed to this version, maybe i should remove the comment...

actually `this.raw.remove` will remove its colliders at the same time from Rust side, but if so then no chance to check with the collider map later (call `co.parent` to match parent), so finally decided use this version.